### ISSUE TITLE
edits are discarded by mastodon web ui

### ISSUE THREAD

**Issue by jhnc** [jlebon/textern#84]
observed behaviour
---

1. access [mastodon](https://github.com/mastodon/mastodon) edit form: `https://`**[instance-name]**`/publish`
2. insert some initial text into form
3. invoke textern (my config = `["xfce4-terminal","--disable-server","--geometry=100x40","-x","vi"]`)
4. insert some text in editor
5. save (`:w`) - new text appears in mastodon web page
6. quit (`:q`) - text in mastodon reverts to initial text

expected behaviour
---

1-5 as observed; 6 should not cause reversion.

notes
---
* text also discarded when using `["xterm","-e","vi"]` or `["mousepad"]`
* observed with mastodon 4.0.2; unsure if affects other versions
* I will also report this to mastodon, in case this is not something textern can fix

**Comment by jhnc** [comment]
Still affects mastodon UI at version 4.2.1

Closed by mastodon as "not planned" whatever that means: https://github.com/mastodon/mastodon/issues/21063

**Comment by chubin** [comment]
I don't think that it is related to Mastodon. In any case, I experience absolutely the same behavior with Github comments (and other Github text areas).

Have you found any workarounds for this bug? Unfortunately, it makes the plugin almost unusable

**Comment by jhnc** [comment]
I can no longer reproduce with my current setup (details at https://github.com/mastodon/mastodon/issues/21063#issuecomment-2646616450), so closing both issues.

@chubin I notice that my firefox sometimes gets into a state where tabs will crash on loading (e.g. imdb, amazon). Generally, if I disable all my plugins and then re-enable them, the crashes stop. Probably won't help  but perhaps worth a try to see if it has any effect on this issue. Could also be an unwanted interaction between firefox plugins (I don't think I'm using the same ones now that I was in 2023).
